### PR TITLE
Python bindings and samples using Pybind11

### PR DESCRIPTION
**Pull request by slovak194**
Hi!

I made it possible to use your lib from python with a thin pybind11 wrapper.
Please, let me know if such a contribution is welcome.

BR, Alex

### CODE REVIEW

**Comment by atakagi-fixstars** [comment]
Hi, @slovak194 

Thank you for your contribution!
I'll check.

**Comment by atakagi-fixstars** [comment]
Hi, @slovak194

Your contribution is welcome.
I reviewed and left some comments on your changes.

Regards,

**Comment by slovak194** [comment]
Great! But where are your comments?

BR, Alex

**Comment by atakagi-fixstars** [comment]
@slovak194

> Great, @atakagi-fixstars! But where are your comments?

Oh, sorry.
My comment were not reflected.

I just posted it.

Regards,

File: pysgm/pysgm.cpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+           py::arg("width") = 612,
+           py::arg("height") = 514,
+           py::arg("disparity_size") = 128,
+           py::arg("input_depth_bits") = 8U,
+           py::arg("output_depth_bits") = 8U,
+           py::arg("inout_type") = sgm::EXECUTE_INOUT::EXECUTE_INOUT_HOST2HOST,
+           py::arg("param") = sgm::StereoSGM::Parameters()
+               )
+      .def(py::init<int, int, int, int, int, int, int, sgm::EXECUTE_INOUT, const sgm::StereoSGM::Parameters &>())
+      .def("execute", [](sgm::StereoSGM &w, unsigned long left_pixels, unsigned long right_pixels, unsigned long dst) {
```
[the remark this answers is not in the document]
Comment (slovak194) [reply pysgm/pysgm.cpp]: Done.

File: sample/pysgm/pysgm_test_raw.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+disp = np.zeros_like(I2)
+
+I1_ptr, _ = I1.__array_interface__['data']
+I2_ptr, _ = I2.__array_interface__['data']
+disp_ptr, _ = disp.__array_interface__['data']
+
+params = pysgm.StereoSGM.Parameters(P1=int(10), P2=int(120), uniqueness=np.float32(0.95), subpixel=False,
+                                    PathType=pysgm.PathType.SCAN_8PATH, min_disp=int(0), LR_max_diff=int(1))
+
+sgm = pysgm.StereoSGM(width=int(612), height=int(514), disparity_size=int(128), input_depth_bits=int(8),
```
[the remark this answers is not in the document]
Comment (slovak194) [reply sample/pysgm/pysgm_test_raw.py]: Done.

File: sample/pysgm/pysgm_test_raw.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                      output_depth_bits=int(8), inout_type=pysgm.EXECUTE_INOUT.EXECUTE_INOUT_HOST2HOST,
+                      param=params)
+
+sgm.execute(I1_ptr, I2_ptr, disp_ptr)
+
+disp_color = cv2.applyColorMap(disp.astype("uint8"), cv2.COLORMAP_JET)
```
[the remark this answers is not in the document]
Comment (slovak194) [reply sample/pysgm/pysgm_test_raw.py:35]: Done.

**Comment by slovak194** [comment]
Hi, @atakagi-fixstars!

I've changed PR according to your suggestions.

BR, Alex

**Comment by atakagi-fixstars** [comment]
Hi, @slovak194

Thanks for your changes!

Let me ask about how to install pysgm.

When I command `pip3 install .` after building project,
pysgm is installed to `/usr/local/lib/python3.7/site-packages/pysgm`.  

But shared library (pysgm.xxx.so) is not included.
I need another procedure to make the path to shared library when I run samples.

Is there better way?

Regards,

**Comment by slovak194** [comment]
> Hi, @slovak194
> 
> Thanks for your changes!
> 
> Let me ask about how to install pysgm.
> 
> When I command `pip3 install .` after building project,
> pysgm is installed to `/usr/local/lib/python3.7/site-packages/pysgm`.
> 
> But shared library (pysgm.xxx.so) is not included.
> I need another procedure to make the path to shared library when I run samples.
> 
> Is there better way?
> 
> Regards,

That's a good point, let me investigate how to do that properly.
I'll come up with proposal soon.

**Comment by slovak194** [comment]
Hi! Please, decide if this PR is relevant.

**Pull request closed without merging**